runners/samza/src/main/java/org/apache/beam/runners/samza/adapter/UnboundedSourceSystem.java

```diff
@@ -363,20 +365,32 @@ public void run() {
       private void updateWatermark() throws InterruptedException {
         final long time = System.currentTimeMillis();
         if (time - lastWatermarkTime > watermarkInterval) {
+          long watermarkMillis = Long.MAX_VALUE;
```

Comment:
This var seems not adding too much value. We should be able to tell the watermark time from nextWatermark.